"Appendable Template" sample added (#58)
liupeirong/MLOpsManufacturing: samples/appendable-template/README.md

        self.output_pipelinedata = output_pipelinedata

    def append_step(self, step_list):
        train_step = PythonScriptStep(
            name="Train Model",
            compute_target=self.compute,
            source_directory=self.env.sources_directory_train,
            script_name=self.env.train_script_path,
            outputs=[self.output_pipelinedata],
            arguments=[
                "--model_name",
                self.pipeline_parameters["model_name"],
                "--step_output",
                self.output_pipelinedata,
                "--dataset_version",
                self.pipeline_parameters["dataset_version"],
                "--data_file_path",
                self.pipeline_parameters["data_file_path"],
                "--caller_run_id",
                self.pipeline_parameters["caller_run_id"],
                "--dataset_name",
                self.env.dataset_name,
            ],
            runconfig=self.config,
            allow_reuse=True,
        )

        if len(step_list) > 0:
            previous_step = step_list[-1]
            train_step.run_after(previous_step)

        step_list.append(train_step)
```

# Getting Started

> ## Modification

Reviewer: I would recommend not to keep both versions in the code. We can link/refer to MLOpsPython for the original non appendable template version.

This sample is based on `diabetes_regression` sample in [`MLOpsPython` repo](https://github.com/microsoft/MLOpsPython/). 

This sample contains minimum files required to run `diabetes_regression_build_train_pipeline.py` locally to build an Azure ML pipeline.

> `diabetes_regression_build_train_pipeline_appendable.py` is a modified version of original `diabetes_regression_build_train_pipeline.py`. It supports "Appendable Tempalte".

Reviewer: ```suggestion
`diabetes_regression_build_train_pipeline.py` is a modified version of the [original version](https://github.com/microsoft/MLOpsPython/blob/master/ml_service/pipelines/diabetes_regression_build_train_pipeline.py). It is adapted to support the "Appendable Template".
```

Three new classes for pipeline steps are added to `ml_service/pipelines/steps` directory.

- `diabetes_regression_build_evaluate_steps.py`
- `diabetes_regression_build_register_steps.py`
- `diabetes_regression_build_train_steps.py`

## Prerequisite for local run

1. Follow ["Development environment setup"](https://github.com/microsoft/MLOpsPython/blob/master/docs/development_setup.md) in `MLOpsPython` repo.

## Running locally

1. [Create an Azure ML workspace](https://docs.microsoft.com/en-us/azure/machine-learning/concept-workspace#-create-a-workspace) in your Azure subscription.

2. Make a copy of [.env.example](local_development/.env.example), place it in the root of this sample, configure the variables, and rename the file to `.env`. You need to configure `SUBSCRIPTION_ID`, `RESOURCE_GROUP` and `WORKSPACE_NAME`.

> 3. Run `diabetes_regression_build_train_pipeline.py` or `diabetes_regression_build_train_pipeline_appendable.py` from `samples/appendable-template` directory.

Reviewer: ```suggestion
3. Run `diabetes_regression_build_train_pipeline.py` from `samples/appendable-template` directory.
```

```bash
export BUILD_BUILDID=$(uuidgen)

> python ml_service/pipelines/diabetes_regression_build_train_pipeline_appendable.py

Reviewer: ```suggestion
python ml_service/pipelines/diabetes_regression_build_train_pipeline.py
```

```

4. In [Azure ML studio](https://ml.azure.com/), click "Pipelines" menu, click "Pipeline endpoints" tab, click the pipeline endpoint you published, and click "Submit" button.

5. In "Set up pipeline run" dialog, select an existing experiment, or create a new experiment, and click "Submit" button.

6. In the pipeline endpoint page, click "Runs" tab, and click the pipeline run you submitted.

## CI/CD in Azure DevOps

1. Clone `MLOpsPython` repo

> 2. Delete the original `diabetes_regression_build_train_pipeline.py` in `ml_service/pipelines` directory of `MLOpsPython` repo.

Reviewer: ```suggestion
2. Replace the `ml_service/pipelines` directory of `MLOpsPython` repo with `samples/appendable-template/ml_service/pipelines` directory of this repo.
```

> 3. Copy `diabetes_regression_build_train_pipeline_appendable.py` to `ml_service/pipelines` directory of `MLOpsPython` repo, and rename it to `diabetes_regression_build_train_pipeline.py`.

Reviewer: ```suggestion

```

> 4. Copy three new classes for pipeline steps to `ml_service/pipelines/steps` directory of `MLOpsPython` repo.

Reviewer: ```suggestion

```

> 5. Follow ["Getting Started with MLOpsPython"](https://github.com/satonaoki/MLOpsPython/blob/master/docs/getting_started.md).

Reviewer: ```suggestion
5. Follow ["Getting Started with MLOpsPython"](https://github.com/microsoft/MLOpsPython/blob/master/docs/getting_started.md).
```
